cellTester/Src/uartRXTask.c

```diff
@@ -0,0 +1,187 @@
+        if (rxBuffer == '\n') {
+            /* A newline character was received, so the input command string is
+            complete and can be processed. */
+            processInput(rxString);
+
+            /* All the strings generated by the input command have been sent.
+            Processing of the command is complete.  Clear the input string ready
+            to receive the next command. */
+            rxIndex = 0;
+            memset(rxString, 0x00, INPUT_BUFFER_SIZE);
+        } else {
+            /* The if() clause performs the processing after a newline character
+            is received.  This else clause performs the processing if any other
+            character is received. */
+
+            if (rxBuffer == '\r') {
+                /* Ignore carriage returns. */
+            } else if (rxBuffer == '\b') {
+                /* Backspace was pressed.  Erase the last character in the input
+                buffer - if there are any. */
+                if (rxIndex > 0) {
+                    rxIndex--;
+                    rxString[rxIndex] = '\0';
+                }
+            } else {
+                /* A character was entered.  It was not a new line, backspace
+                or carriage return, so it is accepted as part of the input and
+                placed into the input buffer.  When a \n is entered the complete
+                string will be passed to the command interpreter. */
+                if (rxIndex < INPUT_BUFFER_SIZE) {
+                    rxString[rxIndex] = rxBuffer;
+                    rxIndex++;
+                } else {
+                    ERROR_PRINT("Rx string buffer overflow\n");
+                }
+            }
+        }
+    }
+}
+
+
+
+void processInput(char* inputString) {
+    bool isNumber = true;
+    bool processedPeriod = false; // Only allow one period for a numerical input
+    char* c = inputString;
+    while (*c != '\0')
+    {
+        DEBUG_PRINT("%c\r\n", *c);
+        // Check to see if its a number
+        if (*c < '0' || *c > '9')
+        {
+            // There can only be one period per decimal
+            if (!processedPeriod && *c == '.')
+            {
+                processedPeriod = true;
+            }
+            else
+            {
+                isNumber = false;
+                break;
+            }
+        }
+        ++c;
+    }
+
+    if (isNumber)
+    {
+        currentTarget = (float)atof(inputString);
+        resetFetDutyCycle();
+    }
+    else if (startCellTestTimeMs == 0 && strcmp(inputString, "start") == 0)
+    {
+        // This tells the GUI to start logging
+        DEBUG_PRINT("start\r\n");
+
+        // Can't start a cellTest if one is in progress
+        startCellTestTimeMs = xTaskGetTickCount();
+    }
+    else if (startCellTestTimeMs != 0 && strcmp(inputString, "stop") == 0)
+    {
+        abortCellTest();
+    }
+    else
+    {
+        DEBUG_PRINT("Unknown command\r\n");
+    }
+}
+
+CellTestStatus_E getCellTestStatus(void)
```

Comment:
This should be moved to a separate file: cellStatus.c

Reply:
Not sure if I agree, we will see what others say but I think its related to the data we are rx'ing over the uart so I think this file is a good spot.